cstore_fdw.c

```diff
@@ -254,6 +240,53 @@ cstore_ddl_event_end_trigger(PG_FUNCTION_ARGS)
 
 
 /*
+ * RegisterCStoreTable calls register_cstore_table UDF to register
+ * given relation id in pg_cstore_tables.
+ */
+static void
+RegisterCStoreTable(Oid relationId)
+{
+	Oid functionOid = InvalidOid;
+	char *schemaName = "public";
+	char *functionName = "register_cstore_table";
+	char *qualifiedFunctionName = NULL;
+	List *qualifiedFunctionNameList = NIL;
+	FuncCandidateList functionList = NULL;
+	int argumentCount = 1;
+	List *argumentNames = NIL;
+	bool expandVariadic = false;
+	bool expandDefaults = false;
+#if PG_VERSION_NUM >= 90400
+	bool missingOK = true;
+#endif
+
+	qualifiedFunctionName = quote_qualified_identifier(schemaName, functionName);
+	qualifiedFunctionNameList = stringToQualifiedNameList(qualifiedFunctionName);
+	functionList = FuncnameGetCandidates(qualifiedFunctionNameList, argumentCount,
+										 argumentNames, expandVariadic,
+										 expandDefaults
+#if PG_VERSION_NUM >= 90400
+										 , missingOK
+#endif
+										 );
+	if (functionList == NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_UNDEFINED_FUNCTION),
+						errmsg("function \"%s\" does not exist", functionName)));
+	}
+	else if (functionList->next != NULL)
+	{
+		ereport(ERROR, (errcode(ERRCODE_AMBIGUOUS_FUNCTION),
+						errmsg("more than one function named \"%s\"", functionName)));
+	}
+
+	functionOid = functionList->oid;
+
+	OidFunctionCall1(functionOid, ObjectIdGetDatum(relationId));
```

Comment:
Can we use `DirectFunctionCall()` for plpgsql functions? (`registrer_cstore_table` is a plpgsql function)